.github/workflows/publish-cli.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and publish the CLI to PyPI
+
+on:
+  workflow_call:
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python client to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing (OIDC)
+      # https://docs.pypi.org/trusted-publishers/using-a-publisher/
+      id-token: write
+    steps:
```

Comment:
its too bad jobs cant share environments so we don have to keep repeating the same 3 commands to set up the repo + poetry